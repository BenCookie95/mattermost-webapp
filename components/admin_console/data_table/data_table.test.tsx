// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import DataTable from './data_table';

describe('components/admin_console/data_table/DataTable', () => {

    const RECORDS_PER_PAGE = 5;

    const baseProps = {
        page: 1,
        startCount: 0,
        endCount: 0,
        total: 0,
        loading: false,

        nextPage: jest.fn(),
        previousPage: jest.fn(),

        rows: [],
        columns: [],
    };

    test('should match snapshot with no items found', () => {
        const wrapper = shallow(
            <DataTable
                {...baseProps}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot while loading', () => {
        const wrapper = shallow(
            <DataTable
                {...baseProps}
                loading={true}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with only one page of results', () => {
        const wrapper = shallow(
            <DataTable
                {...baseProps}
                rows={[
                    {cells: {name: 'Joe Schmoe', team: 'Admin Team'}},
                    {cells: {name: 'Foo Bar', team: 'Admin Team'}},
                    {cells: {name: 'Some Guy', team: 'Admin Team'}},
                ]}
                columns={[
                    {name: 'Name', field: 'name'},
                    {name: 'Team', field: 'team'},
                ]}
                page={0}
                startCount={1}
                endCount={3}
                total={3}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});

