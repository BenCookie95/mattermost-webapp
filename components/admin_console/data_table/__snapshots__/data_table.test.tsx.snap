// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/data_table/DataTable should match snapshot while loading 1`] = `
<Fragment>
  <table
    className="Table__table"
  >
    <DataTableHeader
      columns={Array []}
    />
    <div
      className="Table__loading"
    >
      <LoadingSpinner
        text={null}
      />
      <FormattedMessage
        defaultMessage="Loading"
        id="admin.data_grid.loading"
      />
    </div>
  </table>
</Fragment>
`;

exports[`components/admin_console/data_table/DataTable should match snapshot with no items found 1`] = `
<Fragment>
  <table
    className="Table__table"
  >
    <DataTableHeader
      columns={Array []}
    />
    <div
      className="Table__empty"
    >
      <FormattedMessage
        defaultMessage="No items found"
        id="admin.data_grid.empty"
      />
    </div>
  </table>
</Fragment>
`;

exports[`components/admin_console/data_table/DataTable should match snapshot with only one page of results 1`] = `
<Fragment>
  <table
    className="Table__table"
  >
    <DataTableHeader
      columns={
        Array [
          Object {
            "field": "name",
            "name": "Name",
          },
          Object {
            "field": "team",
            "name": "Team",
          },
        ]
      }
    />
    <DataTableRow
      columns={
        Array [
          Object {
            "field": "name",
            "name": "Name",
          },
          Object {
            "field": "team",
            "name": "Team",
          },
        ]
      }
      key="0"
      row={
        Object {
          "cells": Object {
            "name": "Joe Schmoe",
            "team": "Admin Team",
          },
        }
      }
    />
    <DataTableRow
      columns={
        Array [
          Object {
            "field": "name",
            "name": "Name",
          },
          Object {
            "field": "team",
            "name": "Team",
          },
        ]
      }
      key="1"
      row={
        Object {
          "cells": Object {
            "name": "Foo Bar",
            "team": "Admin Team",
          },
        }
      }
    />
    <DataTableRow
      columns={
        Array [
          Object {
            "field": "name",
            "name": "Name",
          },
          Object {
            "field": "team",
            "name": "Team",
          },
        ]
      }
      key="2"
      row={
        Object {
          "cells": Object {
            "name": "Some Guy",
            "team": "Admin Team",
          },
        }
      }
    />
  </table>
  <div
    className="Table__paging"
  >
    <injectIntl(FormattedMarkdownMessage)
      defaultMessage="{startRecord} - {endRecord} of {totalRecords}"
      id="admin.billing.history.pageInfo"
      values={
        Object {
          "endRecord": 3,
          "startRecord": 1,
          "totalRecords": 3,
        }
      }
    />
    <button
      disabled={true}
      onClick={[Function]}
    >
      <i
        className="icon icon-chevron-left"
      />
    </button>
    <button
      disabled={true}
      onClick={[Function]}
    >
      <i
        className="icon icon-chevron-right"
      />
    </button>
  </div>
</Fragment>
`;
